java-checks/src/main/java/org/sonar/java/checks/AbstractAccessibilityChangeChecker.java

```diff
@@ -145,4 +151,17 @@ protected static boolean isModifyingFieldFromRecord(MethodInvocationTree mit) {
     }
     return Optional.of((IdentifierTree) expression);
   }
+
+  private static boolean identifierPointsToRecord(IdentifierTree classIdentifier) {
+    if (classIdentifier.name().equals("getClass") &&
+      classIdentifier.parent().is(Tree.Kind.METHOD_INVOCATION) &&
+      GET_CLASS_MATCHER.matches((MethodInvocationTree) classIdentifier.parent())) {
+      Tree walker = classIdentifier;
+      while (!walker.is(Tree.Kind.METHOD, Tree.Kind.CONSTRUCTOR)) {
+        walker = walker.parent();
```

Comment:
I'm a bit worried about accessing the parent in a loop. This could lead to performance issues for deeply nested code.